unittests/AST/ASTImporterVisibilityTest.cpp

```diff
@@ -212,6 +262,11 @@ TEST_P(ImportClassesVisibility, ImportAfter) {
 TEST_P(ImportClassesVisibility, ImportAfterImport) {
   TypedTest_ImportAfterImport();
 }
+// EnumDecl.
+TEST_P(ImportEnumsVisibility, ImportAfter) { TypedTest_ImportAfterWithDef(); }
```

Comment:
So the EnumDecls are squashed? How is it different than in case of classes? What is the difference between ImportAfter and ImportAfterWithDef? (Probablye ImportAfterWithDef is not the best name, since I had this question.)

Reply:
It is not possible to forward declare an EnumDecl, it has only definitions. The "WithDef" means that this function is to be used if the thing to import has a definition (this can be a CXXRecordDecl with definition too). These are merged at import or independently created.

Reply:
> It is not possible to forward declare an EnumDecl, it has only definitions.
What about C++11 enum classes? AFAIK they can have fwd decls.